apps/patient/src/components/CouponDetails.tsx

```diff
@@ -0,0 +1,58 @@
+import { Box, HStack, Icon, Text, VStack } from '@chakra-ui/react';
+import { FC, useState } from 'react';
+import { FiInfo } from 'react-icons/fi';
+import { CouponModal } from '../components';
+import { text as t } from '../utils/text';
+import { useOrderContext } from '../views/Main';
+
+export const CouponDetails: FC = () => {
+  const [couponModalOpen, setCouponModalOpen] = useState<boolean>(false);
+  const { order } = useOrderContext();
+
+  const { price, bin, pcn, group, memberId } = order.discountCards[0];
+
+  if (!price || !bin || !pcn || !group || !memberId) {
+    return null;
+  }
+
+  return (
+    <VStack w="full" spacing={4} mx={0}>
+      <CouponModal isOpen={couponModalOpen} onClose={() => setCouponModalOpen(false)} />
+      <Text fontSize="4xl" fontWeight="700" py={0} lineHeight="1">
+        ${price}
+      </Text>
+      <Box bgColor="blue.50" w="full" textAlign="center" p={2} borderRadius="xl">
+        <Text fontWeight="semibold" fontSize="md">
+          {t.showThisCoupon}
+        </Text>
+      </Box>
+      <HStack w="full" align="start">
+        <VStack w="30%" align="start">
+          <Text>{t.bin}</Text>
+          <Text>{t.pcn}</Text>
+          <Text>{t.group}</Text>
+          <Text>{t.memberId}</Text>
```

Comment:
Given that these are actual things, is there a reason we need to pull this from `t`? Can't we just do `BIN`, `PCN` etc